src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs

```diff
@@ -39,65 +39,58 @@ public class WorkItemJob : IQueueJob, IHaveLogger {
                 return JobResult.Success;
 
             if (cancellationToken.IsCancellationRequested) {
-                _logger.Info("Job was cancelled. Abandoning {queueEntryType} work item: {queueEntryId}", queueEntry.Value.Type, queueEntry.Id);
                 await queueEntry.AbandonAsync().AnyContext();
                 return JobResult.CancelledWithMessage($"Abandoning {queueEntry.Value.Type} work item: {queueEntry.Id}");
             }
 
-            Type workItemDataType = null;
+            Type workItemDataType;
             try {
                 workItemDataType = Type.GetType(queueEntry.Value.Type);
             } catch (Exception ex) {
                 await queueEntry.AbandonAsync().AnyContext();
-                return JobResult.FromException(ex, $"Could not resolve {workItemDataType.Name} work item data type.");
+                return JobResult.FromException(ex, $"Abandoning {queueEntry.Value.Type} work item: {queueEntry.Id}: Could not resolve work item data type.");
             }
 
             if (workItemDataType == null) {
                 await queueEntry.AbandonAsync().AnyContext();
-                return JobResult.FailedWithMessage($"Could not resolve {workItemDataType.Name} work item data type.");
+                return JobResult.FailedWithMessage($"Abandoning {queueEntry.Value.Type} work item: {queueEntry.Id}: Could not resolve work item data type.");
             }
 
             object workItemData;
             try {
                 workItemData = await _queue.Serializer.DeserializeAsync(queueEntry.Value.Data, workItemDataType).AnyContext();
             } catch (Exception ex) {
                 await queueEntry.AbandonAsync().AnyContext();
-                return JobResult.FromException(ex, $"Failed to parse {workItemDataType.Name} work item data.");
+                return JobResult.FromException(ex, $"Abandoning {queueEntry.Value.Type} work item: {queueEntry.Id}: Failed to parse {workItemDataType.Name} work item data.");
             }
 
             var handler = _handlers.GetHandler(workItemDataType);
             if (handler == null) {
                 await queueEntry.CompleteAsync().AnyContext();
-                return JobResult.FailedWithMessage($"Handler for type {workItemDataType.Name} not registered.");
+                return JobResult.FailedWithMessage($"Completing {queueEntry.Value.Type} work item: {queueEntry.Id}: Handler for type {workItemDataType.Name} not registered.");
             }
 
             if (queueEntry.Value.SendProgressReports)
-                await _messageBus.PublishAsync(new WorkItemStatus {
-                    WorkItemId = queueEntry.Value.WorkItemId,
-                    Progress = 0,
-                    Type = queueEntry.Value.Type
-                }).AnyContext();
+                await ReportProgress(handler, queueEntry).AnyContext();
```

Comment:
I moved this to a method to remove some repetitive ness but also wrapped it in a try catch with logging. Reporting progress should never blow up job processing.

The question is, should these be a run with retries?